celery/app/base.py

```diff
@@ -63,6 +63,8 @@
     {0}="proj.celeryconfig"
 """
 
+tlocal = threading.local()
```

Comment:
This needs to be moved to live in the `Celery` class instance so that it gets cleaned up by our fixtures during testing and there's no inter-app cross-talk.